src/Module.zig

```diff
@@ -2833,6 +2841,31 @@ fn wrapOptional(self: *Module, scope: *Scope, dest_type: Type, inst: *Inst) !*In
     return self.addUnOp(b, inst.src, dest_type, .wrap_optional, inst);
 }
 
+fn wrapErrorUnion(self: *Module, scope: *Scope, dest_type: Type, inst: *Inst) !*Inst {
+    const err_union = dest_type.castTag(.error_union).?;
```

Comment:
This function needs to check that if `inst.ty == .ErrorSet` that its type is a subset of the error unions error set. Would also be helpful to add a "TODO resolve inferred error set" comment.

Reply:
Does it also have to check that if `inst.ty != .ErrorSet` the payload has to be the same?

Reply:
It has to check that the value can be coerced to the payload type.

Reply:
> This function needs to check that if `inst.ty == .ErrorSet` that its type is a subset of the error unions error set. Would also be helpful to add a "TODO resolve inferred error set" comment.

The hacky way to implement this is to check in `wrapErrorUnion`, but the better way is to implement coercion for error unions like I did in #7092. Ill do it hackily for now, but it should be revisited.